### PR TITLE
Issue #13: Fix getCommandLineArgumentsArray() method signature in cli command

### PR DESCRIPTION
**Pull request by Vinai**

### CODE REVIEW

File: src/lizards-and-pumpkins/bin/triggerTemplateUpdate.php [hunk 1]
```diff
@@ -34,9 +34,6 @@ private function __construct(MasterFactory $factory, CLImate $CLImate)
         $this->setCLImate($CLImate);
     }
 
-    /**
-     * @return RunImport
-     */
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/triggerTemplateUpdate.php]: Should it be PHP 7 return type then?

File: src/lizards-and-pumpkins/bin/triggerTemplateUpdate.php [hunk 2]
```diff
@@ -177,9 +174,6 @@ private function getValidTemplateIds()
         return $templateProjectorLocator->getRegisteredProjectorCodes();
     }
 
-    /**
-     * @return bool
-     */
     protected function isTemplateIdListRequested()
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/triggerTemplateUpdate.php]: Should it be PHP 7 return type then?

**Comment by Vinai** [comment]
Ready for next review

File: src/lizards-and-pumpkins/bin/listUrlKeys.php [hunk 3]
```diff
@@ -48,38 +45,44 @@ public static function bootstrap()
...
         $rawUrlKeyRecords = $this->getDataPoolReader()->getUrlKeysForVersion($version);
-        $urlKeyRecordsForType = array_filter($rawUrlKeyRecords, function ($rawUrlKeyRecord) use ($type) {
-            return $type === 'all' || $rawUrlKeyRecord[self::IDX_TYPE] === $type;
-        });
+        $urlKeyRecordsForType = array_filter(
+            $rawUrlKeyRecords,
+            function ($rawUrlKeyRecord) use ($type) {
+                return $type === 'all' || $rawUrlKeyRecord[self::IDX_TYPE] === $type;
+            }
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/listUrlKeys.php]: Why did you reformatted it?

File: src/lizards-and-pumpkins/bin/listUrlKeys.php [hunk 4]
```diff
@@ -48,38 +45,44 @@ public static function bootstrap()
...
+                    'defaultValue' => 'all',
+                ],
+                'dataVersion' => [
+                    'description'  => 'List url keys for the given catalog data version',
+                    'defaultValue' => 'current',
+                    'required'     => false,
+                ],
             ]
-        ]);
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/listUrlKeys.php]: It is only formatting that changed?

File: src/lizards-and-pumpkins/bin/listUrlKeys.php [hunk 5]
```diff
@@ -88,41 +91,50 @@ protected function execute(CLImate $climate)
...
      * @return string[]
      */
-    private function getFormattedUrlKeysArray($rawUrlKeyRecords)
+    private function getFormattedUrlKeysArray(array $rawUrlKeyRecords) : array
     {
         return $this->getArg('withContext') ?
-            $this->formatUrlKeysWithContext($rawUrlKeyRecords) :
+            $this->formatUrlKeysWithContext(
+                $rawUrlKeyRecords
+            ) :
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/listUrlKeys.php]: Strange re-formatting here too.

File: src/lizards-and-pumpkins/bin/listUrlKeys.php [hunk 6]
```diff
@@ -88,41 +91,50 @@ protected function execute(CLImate $climate)
...
-            return $urlKeyRecord[self::IDX_URL_KEY];
-        }, $rawUrlKeyRecords));
+        return array_unique(
+            array_map(
+                function (array $urlKeyRecord) {
+                    return $urlKeyRecord[self::IDX_URL_KEY];
+                },
+                $rawUrlKeyRecords
+            )
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/listUrlKeys.php]: Re-formatting again.

File: src/lizards-and-pumpkins/bin/listUrlKeys.php [hunk 7]
```diff
@@ -88,41 +91,50 @@ protected function execute(CLImate $climate)
...
-            return sprintf("%-30s\t%s", $urlKeyRecord[self::IDX_URL_KEY], $urlKeyRecord[self::IDX_CONTEXT]);
-        }, $rawUrlKeyRecords));
+        return array_unique(
+            array_map(
+                function (array $urlKeyRecord) {
+                    return sprintf("%-30s\t%s", $urlKeyRecord[self::IDX_URL_KEY], $urlKeyRecord[self::IDX_CONTEXT]);
+                },
+                $rawUrlKeyRecords
+            )
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/listUrlKeys.php]: Same here.

File: src/lizards-and-pumpkins/bin/listUrlKeys.php [hunk 8]
```diff
@@ -151,9 +160,12 @@ private function getDataPoolReader()
...
     {
-        array_map(function ($urlKey) {
-            $this->output($urlKey);
-        }, $formattedUrlKeys);
+        array_map(
+            function ($urlKey) {
+                $this->output($urlKey);
+            },
+            $formattedUrlKeys
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/listUrlKeys.php]: Same here.

File: src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php [hunk 9]
```diff
@@ -21,26 +21,29 @@ public function __construct()
...
+                    'description'  => 'Sort direction (asc|desc)',
+                    'defaultValue' => 'asc',
+                ],
+                'logfile'   => [
+                    'description' => 'Log file',
+                    'required'    => true,
+                ],
             ]
-        ]);
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php]: Same here.

File: src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php [hunk 10]
```diff
@@ -64,15 +67,18 @@ protected function execute(CLImate $climate)
...
-        });
+        usort(
+            $tableData,
+            function (array $rowA, array $rowB) use ($field, $directionalOperator) {
+                $valueA = $this->getComparisonValueFromRow($rowA, $field);
+                $valueB = $this->getComparisonValueFromRow($rowB, $field);
+                $result = $this->threeWayCompare($valueA, $valueB);
+                return $result * $directionalOperator;
+            }
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php]: Same here.

File: src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php [hunk 11]
```diff
@@ -81,48 +87,36 @@ private function sortTableData($tableData, $field, $direction)
...
     {
-        if ($valueA > $valueB) {
-            $result = 1;
-        } elseif ($valueA < $valueB) {
-            $result = -1;
-        } else {
-            $result = 0;
-        }
-        return $result;
+        return $valueA <=> $valueB;
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php]: Yay!

File: src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php [hunk 12]
```diff
@@ -81,48 +87,36 @@ private function sortTableData($tableData, $field, $direction)
...
-        ]);
+        return array_search(
+            $field,
+            [
+                'Handler'     => 'handler',
+                'Count'       => 'count',
+                'Total Sec'   => 'total',
+                'Average Sec' => 'avg',
+            ]
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php]: Re-formatting.

File: src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php [hunk 13]
```diff
@@ -162,28 +152,31 @@ private function getDomainEventHandlerRecordsFromFile($filePath)
...
+    private function buildTableDataFromStats(array $eventHandlerStats) : array
+    {
+        return array_map(
+            function ($handler) use ($eventHandlerStats) {
+                $count = count($eventHandlerStats[$handler]);
+                $sum = array_sum($eventHandlerStats[$handler]);
+                return $this->getTableRow($handler, $count, $sum);
+            },
+            array_keys($eventHandlerStats)
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/reportDomainEventProcessingAvg.php]: Re-formatting.

File: src/lizards-and-pumpkins/bin/runImport.php [hunk 14]
```diff
@@ -66,32 +63,35 @@ private static function enableDebugLogging(
...
+                    'longPrefix'  => 'importImages',
+                    'description' => 'Process images during import',
+                    'noValue'     => true,
+                ],
+                'importFile'    => [
+                    'description' => 'Import XML file',
+                    'required'    => true,
+                ],
+            ]
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/runImport.php]: Re-formatting.

File: src/lizards-and-pumpkins/bin/triggerTemplateUpdate.php [hunk 15]
```diff
@@ -63,26 +60,29 @@ private static function enableDebugLogging(
...
+                    'longPrefix'  => 'list',
+                    'description' => 'List available template IDs',
+                    'noValue'     => true,
+                ],
+                'templateId'    => [
+                    'description' => 'Template ID',
+                    'required'    => false,
+                ],
+            ]
+        );
```
Comment (tim-bezhashvyly) [inline src/lizards-and-pumpkins/bin/triggerTemplateUpdate.php]: Re-formatting.

**Comment by Vinai** [comment]
Thanks for your comments, yes, it's a lot that is only reformatting changes. That is the closest I managed to get PHPStorm to automatically format the way I like. I think it's okay.
Do you have better settings?
I really want to be able to just hit auto-format and be done.

**Pull request merged**